src/components/avatar/avatar.tsx

```diff
@@ -91,20 +128,17 @@ export const EuiAvatar: FunctionComponent<EuiAvatarProps> = ({
     ? '#FFFFFF'
     : '#000000';
 
-  const avatarStyle = {
-    backgroundImage: imageUrl ? `url(${imageUrl})` : 'none',
-    backgroundColor: assignedColor,
-    color: textColor,
-  };
-
   return (
-    <div
-      className={classes}
-      style={avatarStyle}
-      aria-label={name}
-      title={name}
-      {...rest}>
+    <div className={classes} aria-label={name} title={name} {...rest}>
       {optionalInitial}
+      {styles}
```

Comment:
I'm sure I'm naive in here but this feels like something that should just happen and not need to be included? In the badge for example we're just adding the same `styles` stuff into the three possible returns.

Reply:
A bit of obfuscation, but `styles` here is actually a variable storing 2 `<style jsx>` elements (one each for light and dark themes).
Not a requirement to do it like this, and we can likely find a way not to include it all 3 returns, but it does need to be included in some form, just like all the other `<style jsx>` elements.